### PR TITLE
refactor(type): improved attributes-to-props and utilities type

### PR DESCRIPTION
**Pull request by ssi02014**
Release-As: 5.0.4

## What is the motivation for this pull request?

I worked on improving `type` a bit more.

## What is the current behavior?

There is no change in behavior.

## What is the new behavior?

There is no change in behavior.

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: src/attributes-to-props.ts [hunk 1]
```diff
@@ -22,14 +22,16 @@ const valueOnlyInputs = {
...
-export type Props = Record<string, string | boolean> & {
+export type Attributes = Record<PropertyKey, string>;
+
+export type Props = Record<PropertyKey, string | boolean> & {
   dangerouslySetInnerHTML?: {
     __html: string;
   };
   key?: string | number;
-  style?: Record<string, string>;
+  style?: Record<PropertyKey, string>;
```
Comment (ssi02014) [inline src/attributes-to-props.ts:34]: I changed the `keys` generic type of the Record from `string` to the `PropertyKey` provided by Typescript, but what do you think?

File: src/utilities.ts [hunk 2]
```diff
@@ -13,7 +13,10 @@ const RESERVED_SVG_MATHML_ELEMENTS = new Set([
   'font-face-format',
   'font-face-name',
   'missing-glyph',
-]);
+] as const);
+
+type ReservedSvgMathmlElements =
+  typeof RESERVED_SVG_MATHML_ELEMENTS extends Set<infer T> ? T : never;
```
Comment (ssi02014) [inline src/utilities.ts:19]: `Set` is strictly managed, and I declared `type` accordingly.

```ts
type ReservedSvgMathmlElements =
  | 'annotation-xml'
  | 'color-profile'
  | 'font-face'
  | 'font-face-src'
  | 'font-face-uri'
  | 'font-face-format'
  | 'font-face-name'
  | 'missing-glyph';
```

**Review by ssi02014 — COMMENTED** [review]
@remarkablemark 👋
Once again, I worked on improving `type`! I left my intentions as a comment. Thanks!

File: src/attributes-to-props.ts (hunk 1 above)
Comment (remarkablemark) [reply src/attributes-to-props.ts:34]: Looks good

File: src/utilities.ts (hunk 2 above)
Comment (remarkablemark) [reply src/utilities.ts:19]: Looks good

**Comment by ssi02014** [comment]
@remarkablemark 
Thank you for reviewing quickly! 👍👍
I'm glad to be contributing continuously. 😄

**Comment by remarkablemark** [comment]
I appreciate your contributions @ssi02014! 감사합니다

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/1144#issuecomment-1789479907

**Pull request merged**